### PR TITLE
Fixing the graphics bugs introduced since last recent dual quaternion change

### PR DESCRIPTION
**Pull request by samcake**
This pr fixes:
- the avatar rendered big during transitions (when appearing and disappearing from the domain)
- the shadow of the dual quat skinned models

NIssim's PR12516 resolved part of the issues.
I added a fix to the shape key for the dual quat skin filter builder.
And added a DQSkin shading bucket in the shadow map

## TEST PLAN #
Test the above cases, it should be fixed in this pr, broken in current master

in any domain, use pricilia avatar for example: 
[Pricilia](http://mpassets.highfidelity.com/0c954ba0-4d87-4353-b65e-c45509f85658-v1/priscilla.fst)

With another tester also using Pricilia's avatar, make sure that during the fading in and fading out transitions of the avatar (as the other person's come to the same domain or leave it) the avatar is displayed correctly wit the fading effect  (like in current Stable).
In current master (and current RC65 branch) this will show a giant avatar.

Testing in the domain Blue for example,
Enable shadows (in developer/graphics menu)
and make sure that in this pr the shadow casted by this avatar is correct.
in current master it would be broken displaying large black portiions

Thase are the 2 bugs that we know of should be fixed by this pr

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7619/app_12530_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7619/HighFidelity-Beta-PR12530-dc286544c58468b5bfdbb9984518ef665632ef6d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7619/HighFidelity-Beta-PR12530-dc286544c58468b5bfdbb9984518ef665632ef6d.exe

File: libraries/render-utils/src/RenderPipelines.cpp [hunk 1]
```diff
@@ -568,6 +584,22 @@ void initForwardPipelines(ShapePlumber& plumber, const render::ShapePipeline::Ba
...
+    // Dual Quaternion
+    addPipeline(
+        Key::Builder().withMaterial().withSkinned().withDualQuatSkinned(),
+        skinModelDualQuatVertex, modelPixel, nullptr, nullptr);
+    addPipeline(
+        Key::Builder().withMaterial().withSkinned().withTangents().withDualQuatSkinned(),
+        skinModelNormalMapDualQuatVertex, modelNormalMapPixel, nullptr, nullptr);
+    addPipeline(
+        Key::Builder().withMaterial().withSkinned(),
+        skinModelDualQuatVertex, modelSpecularMapPixel, nullptr, nullptr);
```
Comment (SamGondelman) [inline libraries/render-utils/src/RenderPipelines.cpp]: This one doesn't seem right.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7626/app_12530_PR.apk

File: libraries/render-utils/src/RenderPipelines.cpp [hunk 2]
```diff
@@ -568,6 +584,22 @@ void initForwardPipelines(ShapePlumber& plumber, const render::ShapePipeline::Ba
...
+    // Dual Quaternion
+    addPipeline(
+        Key::Builder().withMaterial().withSkinned().withDualQuatSkinned(),
+        skinModelDualQuatVertex, modelPixel, nullptr, nullptr);
+    addPipeline(
+        Key::Builder().withMaterial().withSkinned().withTangents().withDualQuatSkinned(),
+        skinModelNormalMapDualQuatVertex, modelNormalMapPixel, nullptr, nullptr);
+    addPipeline(
+        Key::Builder().withMaterial().withSkinned().withSpecular().withDualQuatSkinned(),
+       skinModelDualQuatVertex, modelSpecularMapPixel, nullptr, nullptr);
```
Comment (SamGondelman) [inline libraries/render-utils/src/RenderPipelines.cpp]: missing a space at the beginning of this line but not a blocker

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7626/HighFidelity-Beta-PR12530-a227cf1edb0051a6a450156a0be229b393d1233f.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7626/HighFidelity-Beta-PR12530-a227cf1edb0051a6a450156a0be229b393d1233f.dmg

**Comment by Qliemillar** [comment]
@samcake Could we get clearer test steps, please? We'd love to run this, but as it stands the test steps are a little too vague and we want to make sure that we are testing correctly and thoroughly :)

**Comment by samcake** [comment]
@Qliemillar i added details, thanks

**Comment by qlimnavarro** [comment]
Currently Testing.

**Comment by qlimnavarro** [comment]
Currently Testing.

**Comment by qlimnavarro** [comment]
Pass - @conklin94122

**Comment by conklin94122** [comment]
@samcake conflicts

**Comment by samcake** [comment]
THe conflict should be fixed
fieeel free to retest to make sure

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7668/app_12530_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7668/HighFidelity-Beta-PR12530-b8fb97c824d77eade4d1f958fdd1e7ca669ae402.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7668/HighFidelity-Beta-PR12530-b8fb97c824d77eade4d1f958fdd1e7ca669ae402.exe

File: libraries/render-utils/src/RenderPipelines.cpp [hunk 3]
```diff
@@ -418,6 +430,10 @@ void initForwardPipelines(ShapePlumber& plumber, const render::ShapePipeline::Ba
     auto skinModelVertex = skin_model_vert::getShader();
     auto skinModelNormalMapVertex = skin_model_normal_map_vert::getShader();
 
+    auto skinModelDualQuatVertex = skin_model_dq_vert::getShader();
+    auto skinModelNormalMapDualQuatVertex = skin_model_normal_map_dq_vert::getShader();
+    auto skinModelNormalMapFadeDualQuatVertex = skin_model_normal_map_fade_dq_vert::getShader();
```
Comment (SamGondelman) [inline libraries/render-utils/src/RenderPipelines.cpp]: Can forward rendered things fade?  None of these pipelines have withFade, so I don't think we need this.

File: libraries/render-utils/src/RenderPipelines.cpp [hunk 4]
```diff
@@ -453,6 +471,8 @@ void initForwardPipelines(ShapePlumber& plumber, const render::ShapePipeline::Ba
     // Skinned Translucents
     addPipeline(Key::Builder().withMaterial().withSkinned().withTranslucent(), skinModelVertex, modelTranslucentPixel);
     addPipeline(Key::Builder().withMaterial().withSkinned().withTranslucent().withTangents(), skinModelNormalMapVertex, modelTranslucentPixel);
+    addPipeline(Key::Builder().withMaterial().withSkinned().withTranslucent().withDualQuatSkinned(), skinModelDualQuatVertex, modelTranslucentPixel);
+    addPipeline(Key::Builder().withMaterial().withSkinned().withTranslucent().withTangents().withDualQuatSkinned(), skinModelNormalMapFadeDualQuatVertex, modelTranslucentPixel);
```
Comment (SamGondelman) [inline libraries/render-utils/src/RenderPipelines.cpp]: This should use skinModelNormalMapDualQuatVertex, not skinModelNormalMapFadeDualQuatVertex.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7684/app_12530_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7684/HighFidelity-Beta-PR12530-fc8f1c1e65825f1d15047a4c6afe1f0e12635d43.exe

**Pull request merged**